src/browser/cached_resource_fetcher.ts

```diff
@@ -13,85 +13,213 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
-import {app, resourceFetcher} from 'electron'; // Electron app
-import {stat, mkdir} from 'fs';
-import {join, parse} from 'path';
-import {parse as parseUrl} from 'url';
+import { app, net, ClientResponse, nativeImage, NativeImage } from 'electron'; // Electron app
+import { stat, mkdir, readFile, writeFile, unlink } from 'fs';
+import { join, parse } from 'path';
+import { parse as parseUrl } from 'url';
 import {createHash} from 'crypto';
-import {isURL, isURI, uriToPath} from '../common/regex';
+import * as log from './log';
 
-let appQuiting: Boolean = false;
+import { isURL, isICO, isImageFile, Patterns } from '../common/regex';
+import { getDataURL } from './assets';
 
-app.on('quit', () => { appQuiting = true; });
+let appQuitting: boolean = false;
+app.on('quit', () => { appQuitting = true; });
+
+const ERR_QUITTING: string = 'Runtime is exiting';
+
+export class FetchResponse {
+    constructor(success: boolean = false, data?: string, isImageData: boolean = false) {
```

Comment:
Optional arg in front of a default?

Reply:
as discussed, I don't really need it to be `false`; it just needs to be falsy to `undefined` is sufficient. Therefore, I'm OK with changing it to:
```ts
    constructor(success: boolean = false, data?: string, isImageData?: boolean) {
```